tensorflow/python/compiler/tensorrt/__init__.py

```diff
@@ -19,5 +19,74 @@
 from __future__ import print_function
 
 # pylint: disable=unused-import,line-too-long
+from tensorflow.compiler.tf2tensorrt import _pywrap_py_utils
 from tensorflow.python.compiler.tensorrt import trt_convert as trt
+from tensorflow.python.compiler.tensorrt import utils as trt_utils
+from tensorflow.python.platform import tf_logging as logging
 # pylint: enable=unused-import,line-too-long
+
+
+if not _pywrap_py_utils.is_tensorrt_enabled():
+  """Check that TensorFlow has been built with TensorRT support enabled.
+  Raises:
+    RuntimeError: if the TensorFlow has not be built with TensorRT support.
+  """
+  logging.error(
+      "Tensorflow needs to be built with TensorRT support enabled to allow "
+      "TF-TRT to operate.")
+
+  raise RuntimeError("Tensorflow has not been built with TensorRT support.")
+
+else:
+  """Check compatibility of TensorRT version.
+  Raises:
+    RuntimeError: if the TensorRT library version is incompatible.
+  """
+
+  linked_version = _pywrap_py_utils.get_linked_tensorrt_version()
+  loaded_version = _pywrap_py_utils.get_loaded_tensorrt_version()
+
+  logging.info("Linked TensorRT version: %s", str(linked_version))
+  logging.info("Loaded TensorRT version: %s", str(loaded_version))
+
+
+  def raise_trt_version_deprecated(version_type, trt_version):
+    assert version_type in [
+        "linked", "loaded"
+    ], ("Incorrect value received for version_type: %s. Accepted: ['linked', "
+        "'loaded']") % version_type
+
+    logging.error(
+        "The {version_type} version of TensorRT: `{trt_version}` has now "
+        "been removed. Please upgrade to TensorRT 7 or more recent.".format(
+            version_type=version_type,
+            trt_version=trt_utils.version_tuple_to_string(trt_version)))
+
+    raise RuntimeError("Incompatible %s TensorRT versions" % version_type)
+
+
+  if not trt_utils.is_linked_tensorrt_version_greater_equal(7, 0, 0):
+    raise_trt_version_deprecated("linked", linked_version)
+
+  if not trt_utils.is_loaded_tensorrt_version_greater_equal(7, 0, 0):
+    raise_trt_version_deprecated("loaded", loaded_version)
+
+  if (loaded_version[0] != linked_version[0] or
+      not trt_utils.is_loaded_tensorrt_version_greater_equal(*linked_version)):
+    logging.error(
+        "Loaded TensorRT %s but linked TensorFlow against TensorRT %s. A few "
+        "requirements must be met:\n"
+        "\t-It is required to use the same major version of TensorRT during "
+        "compilation and runtime.\n"
+        "\t-TensorRT does not support forward compatibility. The loaded "
+        "version has to be equal or more recent than the linked version.",
```

Comment:
Let's s/to be equal or more recent than/to be not later than/

Reply:
Fixed